lib/rules/semi-spacing.js

```diff
@@ -104,22 +104,52 @@ module.exports = function(context) {
 
             if (hasLeadingSpace(token)) {
                 if (!requireSpaceBefore) {
-                    context.report(node, location, "Unexpected whitespace before semicolon.");
+                    context.report({
```

Comment:
@gyandeeps can you point me at one example you like? Or tell me how it would be different from the current `checkSemicolonSpacing` function.